packages/auth/amplify_auth_cognito_dart/lib/src/auth_plugin_impl.dart

```diff
@@ -993,6 +996,46 @@ class AmplifyAuthCognitoDart extends AuthPluginInterface
         .result;
   }
 
+  @override
+  Future<CognitoDevice> fetchCurrentDevice() async {
+    final tokens = await stateMachine.getUserPoolTokens();
+    final deviceSecrets = await _deviceRepo.get(tokens.username);
+    final deviceKey = deviceSecrets?.deviceKey;
+    if (deviceSecrets == null || deviceKey == null) {
+      throw const DeviceNotTrackedException();
+    }
+
+    late GetDeviceResponse resp;
```

Comment:
nit: IMO the abbreviation doesn't add much value here.
```suggestion
    late GetDeviceResponse response;
```

Reply:
added